packages/web/src/components/CheckboxField/_CheckboxField.scss

```diff
@@ -0,0 +1,90 @@
+@use 'theme';
+@use '../../tools/typography';
+
+.CheckboxField {
+    max-width: 100%;
+}
+
+.CheckboxField__label {
+    @include typography.generate(theme.$label-typography);
+
+    margin-bottom: theme.$spacing;
+    color: theme.$label-color-default;
+}
+
+.CheckboxField__label--hidden {
+    position: absolute;
+    overflow: hidden;
+    width: 1px;
+    height: 1px;
+    padding: 0;
+    margin: -1px;
+    clip: rect(0, 0, 0, 0);
+    white-space: nowrap;
+    border: 0;
+}
+
+.CheckboxField__label--required::after {
+    content: '*';
+    color: theme.$label-required-color;
+}
+
+.CheckboxField__input {
+    appearance: none;
+    width: theme.$input-width;
+    height: theme.$input-width;
+    margin-right: 6px;
+    vertical-align: top;
+    background-image: theme.$input-unchecked-background-image;
+    background-position: center;
+    background-size: contain;
+    background-repeat: no-repeat;
+
+    &:checked {
+        background-image: theme.$input-checked-background-image;
+    }
+
+    &:indeterminate {
+        background-image: theme.$input-indeterminate-background-image;
+    }
+}
+
+.CheckboxField__message {
+    @include typography.generate(theme.$message-typography);
+
+    margin-top: theme.$spacing;
+    margin-left: theme.$message-offset;
+    color: theme.$message-color-default;
+}
+
+.CheckboxField--error > .CheckboxField__input,
+.CheckboxField--error:hover > .CheckboxField__input,
+.CheckboxField--error > .CheckboxField__input:focus {
+    border-color: theme.$input-border-color-error;
+}
+
+.CheckboxField--error > .CheckboxField__message {
+    color: theme.$message-color-error;
+}
+
+.CheckboxField--disabled > .CheckboxField__label {
+    color: theme.$label-color-disabled;
+}
+
+.CheckboxField .CheckboxField__input:disabled,
+.CheckboxField:hover .CheckboxField__input:disabled,
+.CheckboxField--disabled .CheckboxField__input {
+    background-image: theme.$input-unchecked-disabled-background-image;
+
+    &:checked {
+        background-image: theme.$input-checked-disabled-background-image;
+    }
+
+    &:indeterminate {
```

Comment:
Could we have this state in our examples too?

Reply:
As I know, for this state I must use Javascript to set it (https://developer.mozilla.org/en-US/docs/Web/CSS/:indeterminate#checkbox_radio_button). I think I could add it.

Reply:
Added.